### PR TITLE
upgrade Ember-cli to 2.12.3

### PR DESCRIPTION
**Pull request by helloWorldCindy**
**This project uses [semver](http://semver.org), please check the scope of this pr:**

- [ ] #none# - documentation fixes and/or test additions
- [x] #patch# - backwards-compatible bug fix
- [ ] #minor# - adding functionality in a backwards-compatible manner
- [ ] #major# - incompatible API change

# CHANGELOG

Upgrade Ember-cli to 2.12.3

### CODE REVIEW

File: .npmignore [hunk 1]
```diff
@@ -10,6 +11,7 @@
 .ember-cli
 .eslintignore
 .eslintrc
```
Comment (notmessenger) [inline .npmignore]: Remove

File: .npmignore [hunk 2]
```diff
@@ -10,6 +11,7 @@
...
+.eslintrc.js
 .github
 .gitignore
 .jshintrc
```
Comment (notmessenger) [inline .npmignore]: Remove

File: .travis.yml [hunk 3]
```diff
@@ -45,6 +30,7 @@ matrix:
   - env: EMBER_TRY_SCENARIO=ember-release
   - env: EMBER_TRY_SCENARIO=ember-beta
   - env: EMBER_TRY_SCENARIO=ember-canary
+  - node_js: 'stable'
```
Comment (notmessenger) [inline .travis.yml]: Do we want this @juwara0 ?

File: tests/dummy/app/templates/application.hbs [hunk 4]
```diff
@@ -0,0 +1,5 @@
+{{!-- The following component displays Ember's default welcome message. --}}
```
Comment (notmessenger) [inline tests/dummy/app/templates/application.hbs]: Remove this file

File: package.json [hunk 5]
```diff
@@ -23,23 +23,24 @@
...
-    "ember-cli": "^2.11.0",
-    "ember-cli-app-version": "^2.0.1",
+    "broccoli-asset-rev": "^2.4.5",
+    "ember-cli": "2.12.3",
     "ember-cli-chai": "^0.3.2",
     "ember-cli-dependency-checker": "^1.3.0",
     "ember-cli-htmlbars": "^1.1.1",
-    "ember-cli-htmlbars-inline-precompile": "0.3.6",
+    "ember-cli-htmlbars-inline-precompile": "^0.3.6",
     "ember-cli-inject-live-reload": "^1.6.1",
```
Comment (notmessenger) [inline package.json]: Change to `^1.4.1`

**Comment by notmessenger** [comment]
Remove `ENV.rootURL = '/'` from test section of _ tests/dummy/config/environment.js_

File: .travis.yml (hunk 3 above)
Comment (dafortin) [inline .travis.yml]: Do we want to keep `release`, `beta` and `canary`?

File: .npmignore (hunk 1 above)
Comment (helloWorldCindy) [inline .npmignore]: Fixed.

File: .npmignore (hunk 2 above)
Comment (helloWorldCindy) [inline .npmignore]: Fixed.

File: tests/dummy/app/templates/application.hbs (hunk 4 above)
Comment (helloWorldCindy) [inline tests/dummy/app/templates/application.hbs]: Fixed.

File: package.json (hunk 5 above)
Comment (helloWorldCindy) [inline package.json]: Fixed.

Comment (helloWorldCindy) [inline package.json]: Fixed.

File: package.json [hunk 6]
```diff
@@ -23,23 +23,24 @@
...
     "ember-cli-chai": "^0.3.2",
     "ember-cli-dependency-checker": "^1.3.0",
     "ember-cli-htmlbars": "^1.1.1",
-    "ember-cli-htmlbars-inline-precompile": "0.3.6",
-    "ember-cli-inject-live-reload": "^1.6.1",
+    "ember-cli-htmlbars-inline-precompile": "^0.3.6",
+    "ember-cli-inject-live-reload": "^1.4.1",
     "ember-cli-mocha": "0.13.2",
+    "ember-cli-shims": "^1.0.2",
     "ember-cli-test-loader": "^1.1.1",
```
Comment (juwara0) [inline package.json]: This can be removed

File: package.json [hunk 7]
```diff
@@ -23,23 +23,24 @@
   "author": "Matthew Dahl (https://github.com/sandersky)",
   "license": "MIT",
   "devDependencies": {
-    "broccoli-asset-rev": "^2.5.0",
-    "ember-cli": "^2.11.0",
-    "ember-cli-app-version": "^2.0.1",
+    "broccoli-asset-rev": "^2.4.5",
+    "ember-cli": "^2.12.3",
```
Comment (juwara0) [inline package.json]: This should not have the float `^`.

Comment (juwara0) [inline package.json]: Fixed

File: package.json (hunk 6 above)
Comment (juwara0) [inline package.json]: Fixed

File: .travis.yml (hunk 3 above)
Comment (juwara0) [inline .travis.yml]: I have fixed this.

File: tests/dummy/config/environment.js [hunk 8]
```diff
@@ -1,13 +1,17 @@
+/* eslint-env node */
 module.exports = function (environment) {
```
Comment (notmessenger) [inline tests/dummy/config/environment.js]: `ENV.rootURL = '/'` should be taken out of `test` configuration

**Comment by notmessenger** [comment]
`ENV.rootURL = '/'` should be taken out of `test` configuration

File: tests/dummy/config/environment.js (hunk 8 above)
Comment (juwara0) [inline tests/dummy/config/environment.js]: Fixed

**Comment by notmessenger** [comment]
👍

**Pull request merged**